### PR TITLE
Fix batch normalization statistics in StochasticWeightAveraging

### LINKED ISSUE #14866: Potential bug in StochasticWeightAveraging causing incorrect batchnorm statistics
### First check

- [X] I'm sure this is a bug.
- [X] I've added a descriptive title to this bug.
- [X] I've provided clear instructions on how to reproduce the bug.
- [X] I've added a code sample.
- [X] I've provided any other important info that is required.

### Bug description

I recently noticed that my model (ResNet trained on a subset of ImageNet) has very bad performance (accuracy ~0) after training with StochasticWeightAveraging. The model performs well (accuracy ~.75) without using StochasticWeightAveraging. After some while, I found that the problem seems to be incorrect batchnorm stats and located the cause in this line:

https://github.com/Lightning-AI/lightning/blob/c77d4a8394a307992b27fe935831b2fc83b5ce6c/src/pytorch_lightning/callbacks/stochastic_weight_avg.py#L299

comparing with the pytorch SWA implementation, which has
https://github.com/pytorch/pytorch/blob/57bffc3a8e4fee0cce31e1ff1f662ccf7b16db57/torch/optim/swa_utils.py#L153
```
module.momentum = None
```
Setting ``module.momentum`` to ``None`` or ``0.`` behaves differently, according to 
https://github.com/pytorch/pytorch/blob/b136f3f310aa01a8b3c1e63dc0bfda8fd2234b06/torch/nn/modules/batchnorm.py#L152-L155
It seems that ``module.momentum = None`` accumulates an averaged statistics, which is the behavior we want.

After changing to ``
module.momentum = None
`` my model's performance became normal.

### How to reproduce the bug

```python
I'm not sure if others can reproduce this bug, but we can first discuss based on the code.
```

### Error messages and logs

_No response_

### Important info

```
#- Lightning Component: Callback
#- PyTorch Lightning Version: 1.7.7
#- PyTorch Version: 1.12.1
#- Python version: 3.10
```

### More info

_No response_

### PR DESCRIPTION
**Pull request by zxvix**
## What does this PR do?

Fixed incorrect batch normalization statistics calculation in `StochasticWeightAveraging` callback.

Fixes #14866

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/callbacks/stochastic_weight_avg.py [hunk 1]
```diff
@@ -298,7 +298,7 @@ def reset_batch_norm_and_save_state(self, pl_module: "pl.LightningModule") -> No
                 dtype=module.running_var.dtype,
             )
             self.momenta[module] = module.momentum
-            module.momentum = float()
+            module.momentum = None
```
Comment (awaelchli) [inline src/pytorch_lightning/callbacks/stochastic_weight_avg.py:301]: This bug was introduced here: https://github.com/Lightning-AI/lightning/pull/13860/files

We should update the type of `self.momenta` accordingly, where values can take Optional[float]

**Review by awaelchli — APPROVED** [review]
